### DISCUSSION TITLE
Collections with self referencing relationships cannot be removed

### DISCUSSION THREAD

**Discussion by rlora** [directus/directus#24894] [Feature Requests]
### Describe the Bug

We have a basic use case that we're currently implementing with a workaround, but we couldn't find a clear reference whether this is currently supported in Directus or not.

We have a `pages` collection that represents pages in a website. A page can either be a root page or have an optional parent. This allow us to model a hierarchical structure where we can display menus and submenus.

Most of the model works as expected but the collection cannot be removed without removing the relation field first. Also we cannot configure the behavior "SET NULL" to the relationship.

The model is as follows (simplified):

- pages
  - id (pk)
  - title (required, string)
  - parent (optional, string, self-referencing) 

Some sample data:

```
('1', 'Use cases', null)
('2', 'Use case A', '1')
('3', 'Use case B', '1')
('4', 'About', null)
```

### To Reproduce

Create a collection as the one described in above. Try to delete the collection.

### Errors Shown

The remove action fails with the following error in the Directus console.
![Screen Shot 2022-06-09 at 11 34 50 AM](https://user-images.githubusercontent.com/1533339/172889575-d5785d97-9bc5-4320-890e-96bca6abcab9.png)

### What version of Directus are you using?

9.12.1

### What version of Node.js are you using?

16.15.0

### What database are you using?

SQLite 3.38.x

### What browser are you using?

Firefox

### How are you deploying Directus?

Running locally via docker compose

**Comment by licitdev** [comment]
`await trx.schema.dropTable(collectionKey);` was shifted to the start of the transaction instead of after relational fields are deleted. Thus resulting in the error as the table no longer exists. 
Ref: https://github.com/directus/directus/pull/11279/commits/24b96dc2c01f53a8134ba1cd84f9d4e39dd4a380

**Comment by willemmulder** [comment]
Just as a note from #14717 : this also happens on Directus Cloud and when we 'force' a field to become a foreign key by mentioning it as a foreign field in e.g. a tree view field.

**Comment by rijkvanzanten** [comment]
> Just as a note from #14717 : this also happens on Directus Cloud and when we 'force' a field to become a foreign key by mentioning it as a foreign field in e.g. a tree view field.

(On the community tier, as that uses SQLite. Standard/enterprise are unaffected 👍🏻 )

**Comment by rijkvanzanten** [comment]
Linear: [ENG-246](https://linear.app/directus/issue/ENG-246/collections-with-self-referencing-relationships-cannot-be-removed)
